doc/leak_tracking/DETECT.md

```diff
@@ -0,0 +1,165 @@
+
+Coming soon! See https://github.com/flutter/devtools/issues/3951.
+
+The text below is under construction.
+
+# Detect Memory Leaks
+
+This page describes how to auto-detect certain types of memory leaks.
+Read more about leak tracking in [overview](OVERVIEW.md).
+
+## Quick start to track leaks for Flutter
+
+### Test cover with `testWidgets`
+
+The Flutter test method `testWidgets` can be configured to track and detect leaks
+from all instrumented classes. To enable leak tracking for your entire test suite
+and to configure leak tracking settings, add or modify your
+[`test/flutter_test_config.dart`](https://api.flutter.dev/flutter/flutter_test/flutter_test-library.html)
+file:
+
+```dart
+import 'dart:async';
+
+import 'package:leak_tracker_flutter_testing/leak_tracker_flutter_testing.dart';
+
+...
+
+FutureOr<void> testExecutable(FutureOr<void> Function() testMain) {
+  LeakTesting.enable();
+  LeakTesting.settings = LeakTesting.settings
+    .withIgnored(createdByTestHelpers: true);
+
+  ...
+
+  return testMain();
+}
+```
+
+You can also adjust leak tracking settings for individual tests:
+
+```dart
+  testWidgets('Images are rendered as expected.',
+  // TODO(polina-c): make sure images are disposed, https://github.com/polina-c/my_repo/issues/141
+  experimentalLeakTesting: LeakTesting.settings.withIgnored(classes: ['Image']),
```

Comment:
The formatting here seems off, and is this really the API that we want to have?

In particular, the parameter name seems strange, and which `Image` classes are we ignoring? The one from dart:ui? The one from package:flutter/widgets.dart? The one from package:image/image.dart? The one from my own application?

Reply:
It will ignore all leaks for classes named 'Images'.

In reality, while debugging tests, there is no confusion about it. Normally, it is clear what image is leaking.

Requiring fully qualified name will impact velocity of debugging and conciseness of code. If we ever get complains about it, we will improve it.